src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/entity/CreatureBuildListener.java

```diff
@@ -35,24 +34,29 @@ public void onCreatureSpawn(CreatureSpawnEvent event) {
                 && event.getSpawnReason() != CreatureSpawnEvent.SpawnReason.BUILD_WITHER) {
             return;
         }
+
         Location location = event.getLocation();
         Collection<Block> removedBlocks;
-        if (event.getEntityType() == EntityType.IRON_GOLEM) {
-            // Add the iron blocks
-            removedBlocks = TShapedBlockPattern.getMatchingBlocks(Material.IRON_BLOCK, location);
-            // Add the carved pumpkin head
-            removedBlocks.add(location.getBlock().getRelative(BlockFace.UP, 2));
-        } else if (event.getEntityType() == EntityType.SNOWMAN) {
-            // Add the two snow blocks and the carved pumpkin head
-            Block base = location.getBlock();
-            removedBlocks = Arrays.asList(base, base.getRelative(BlockFace.UP), base.getRelative(BlockFace.UP, 2));
-        } else if (event.getEntityType() == EntityType.WITHER) {
-            // Add the soul sand and wither skulls
-            removedBlocks = WitherBuildPattern.getMatchingBlocks(location);
-        } else {
-            // This should not happen as we checked the SpawnReason earlier
-            // This return statement is just to make the compiler happy
-            return;
+        switch (event.getEntityType()) {
+            case IRON_GOLEM -> {
+                // Add the iron blocks
+                removedBlocks = TShapedBlockPattern.getMatchingBlocks(Material.IRON_BLOCK, location);
+                // Add the carved pumpkin head
+                removedBlocks.add(location.getBlock().getRelative(BlockFace.UP, 2));
+            }
+            case SNOWMAN -> {
+                // Add the two snow blocks and the carved pumpkin head
+                Block base = location.getBlock();
+                removedBlocks = Arrays.asList(base, base.getRelative(BlockFace.UP), base.getRelative(BlockFace.UP, 2));
+            }
+            case WITHER ->
+                // Add the soul sand and wither skulls
+                removedBlocks = WitherBuildPattern.getMatchingBlocks(location);
+            default -> {
+                // This should not happen as we checked the SpawnReason earlier
+                // This return statement is just to make the compiler happy
```

Comment:
Might be worth adding a WARNING log here, as in "unexpected entity type X spawned through an unknown build pattern." or something.